dotCMS/src/main/java/com/dotmarketing/startup/StartupTasksExecutor.java

```diff
@@ -10,346 +13,208 @@
-						}
-
-						Logger.info(this, "Database upgraded to version: "
-								+ taskId);
-						HibernateUtil.closeAndCommitTransaction();
-
-					}
-				} catch (NumberFormatException e) {
-					Logger
-							.error(
-									this,
-									"Class "
-											+ name
-											+ " has invalid name or shouldn't be in the tasks package.");
-
-				}
-			}
-	
-			ReindexThread.unpause();
-			
-		} catch (Throwable e) {
-			HibernateUtil.rollbackTransaction();
-      if(Config.getBooleanProperty("SYSTEM_EXIT_ON_STARTUP_FAILURE", true)){
-        e.printStackTrace();
-        System.exit(1);
-      }
-			throw new DotDataException("Unable to execute startup task : ",e);
-		} finally {
-			// This will commit the changes and close the connection
-			HibernateUtil.closeSession();
-			
-			// DOTCMS-4352
-			try {
-				if(connection != null && !connection.isClosed()){
-					Statement s1 = connection.createStatement();
-					if(DbConnectionFactory.isMySql()){
-						s1.execute(commit);
-					}
-					if(!connection.getAutoCommit()){
-						connection.commit();
-					}
-					connection.close();
-				}
-			} catch (Exception e) {
-				Logger.debug(StartupTasksExecutor.class, "Exception: "
-						+ e.getMessage(), e);
-				throw new DotDataException(
-						"Exception finishing upgrade tasks: " + e.getMessage(),
-						e);
-			} finally {
-
-				CloseUtils.closeQuietly(connection);
-			}
-			Logger.info(this, "Finishing upgrade tasks.");
-		}
-
-	}
+    /**
+     * This will create the db version table if it does not already exist
+     * @return
+     */
+	private boolean insureDbVersionTable() {
+
+        try {
+            currentDbVersion();
+            return true;
+
+        } catch (Exception e) {
+            return createDbVersionTable();
+        }
+
+    }
+    
+    /**
+     * Runs with a separate DB connection
+     * @return
+     */
+    private int currentDbVersion() {
+        try (Connection conn = DbConnectionFactory.getDataSource().getConnection()) {
+            Map<String,Object> results =  new DotConnect().setSQL(select).loadObjectResults().get(0);
+            return  ((Long) results.get("count_versions"))>0
+                        ? ((int) results.get("db_version"))
+                        : 0;
+
+        } catch (Exception e) {
+            throw new DotRuntimeException(e);
+        }
+    }
+    
+    /**
+     * Runs with a separate DB connection
+     * @return
+     */
+    private boolean createDbVersionTable() {
+
+        try (Connection conn = DbConnectionFactory.getDataSource().getConnection()) {
+            new DotConnect().setSQL(createTableSQL()).loadResult(conn);
+            return true;
+
+        } catch (Exception e) {
+            Logger.debug(this.getClass(), e.getMessage(),e);
+            throw new DotRuntimeException(e);
+        }
+
+    }
+    
+    
+    public void executeStartUpTasks() throws DotDataException {
+        
+        Logger.debug(this.getClass(), "Running Startup Tasks");
```

Comment:
I think it's redundant because there is a Logger.info just below with the same info